accumulator/transform.go

```diff
@@ -1,5 +1,65 @@
 package accumulator
 
+func makeCollapse(dels []uint64, rootPresent bool, r uint8, numLeaves, nextNumLeaves uint64, forestRows uint8) []arrow {
+	delRemains := len(dels)%2 != 0
+	if rootPresent == delRemains {
```

Comment:
@kcalvinalvin Did you see this line? This makes sure the function only proceeds in the desired configurations of rootPresent and delRemains, keeping the old behaviour. I chose to do it this way, because the loop is already complicated, so I preferred to have branching in the functions, which are pure.

Reply:
oh I see what's going on now. Yeah I did miss the line :/